tests/cases/compiler/partialType2.ts

```diff
@@ -0,0 +1,25 @@
+interface State {
+    name: string;
+    length: number;
+    foo?: number;
+}
+
+function doSomething1<T>(x: T) {
+    let y: partial T = <any>null;
+    y = x; // Should be OK
+    x = y; // Error
+}
+
+function doSomething2<T extends State>(x: T) {

```

Comment:
Or maybe I misread your summary and you were talking about assignability rules. In that case are there *any* valid inferences to make with `partial T` ?